exporter/signalfxexporter/exporter.go

```diff
@@ -230,10 +230,9 @@ func (se *signalfxExporter) startLogs(_ context.Context, host component.Host) er
 func (se *signalfxExporter) createClient(host component.Host) (*http.Client, error) {
 	se.config.HTTPClientSettings.TLSSetting = se.config.IngestTLSSettings
 
-	if se.config.HTTPClientSettings.MaxIdleConns == nil {
+	if se.config.MaxConnections != 0 {
+		se.logger.Warn("You are using the deprecated `max_connections` option that will be removed in the next release; use `max_idle_conns` and `max_idle_conns_per_host` instead: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/signalfxexporter#advanced-configuration")
 		se.config.HTTPClientSettings.MaxIdleConns = &se.config.MaxConnections
```

Comment:
`MaxIdleConns` should have precedence over the deprecated `MaxConnections`, so we need an additional `se.config.HTTPClientSettings.MaxIdleConns != nil` condition here. Same goes to `MaxIdleConnsPerHost`

Reply:
Makes sense! On it.